jruby-gradle-base-plugin/src/main/groovy/com/github/jrubygradle/internal/mavengem/Handler.java

```diff
@@ -0,0 +1,57 @@
+package com.github.jrubygradle.internal.mavengem;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.net.URLStreamHandler;
+
+public class Handler extends URLStreamHandler {
+
+    public static String KEY = "java.protocol.handler.pkgs";
+    public static String PKG = "com.github.jrubygradle.internal";
+    private final String MAVEN_RELEASES = "/maven/releases";
+    private final int MAVEN_RELEASES_LEN = MAVEN_RELEASES.length();
+
+    public synchronized static boolean isMavenGemProtocolRegistered() {
+        return System.getProperties().contains(KEY);
+    }
+
+    public synchronized static boolean registerMavenGemProtocol() {
+        if (System.getProperties().contains(KEY)) {
+            String current = System.getProperty(KEY);
+            if (!current.contains(PKG)) {
+                System.setProperty(KEY, current + "|" + PKG);
+            }
+        }
+        else {
+            System.setProperty(KEY, PKG);
+        }
+        try {
+            URL url = new URL("mavengem:https://rubygems.org/maven/releases/ping");
```

Comment:
the ping works offline. and I am not sure yet if this is needed. the idea is to use the mavengem protocol ones to register it. as this code runs from the classloader which contains the mavengem protocol code. doing it the first time from the plugin classloader will not find the protocol code via just the System.properties.

add a comment here